### DISCUSSION TITLE
Custom app service not visible in consul and dashboard app service list

### DISCUSSION THREAD

**Discussion by majidalikhan8baader** [edgexfoundry/community#112] [Q&A] [answered]
We have created a custom application service. It runs and works fine with what we need for the application but we can not see the custom application service in the EdgeX dashboard UI and also the configurations are not presented in the Consul as well. 

We want it to be present there so that we can update the configurations from the consul itself rather than doing it in the file. 

Any ideas about this? Any documentation or examples? I have tried searching it in the docu but I couldn't. Maybe I am missing something. 

So thanks in advance.

**Comment by lenny-goodell** [comment, marked as answer]
The EdgeX UI only shows instances of App Service Configurable. Custom app services will not show on the EdgeX UI.

As for configuration in Consul, you must use the -cp/--configProvider command line flag.
See docs here: https://docs.edgexfoundry.org/2.3/microservices/configuration/CommonCommandLineOptions/#config-provider
Example in docker file here:  https://github.com/edgexfoundry/app-service-configurable/blob/main/Dockerfile#L61

**Reply by majidalikhan8baader** [reply]
Thanks for the answer. Will take more look into it. Although, is there a possible plan to make the custom application services visible in the EdgeX UI in future?

**Reply by majidalikhan8baader** [reply]
Unfortunately, I am removing it as a marked-by-answer as I want to continue the discussion as I was unable to see the config in the consul with the mentioned changes. 

Does the Consul part also work for custom application services or only for the App Service Configurable? Because it is for my docker file, it ain't showing up with the mentioned changes as well. Below is my docker file example.

```
#
# Copyright (c) 2022 Intel Corporation
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

#build stage
ARG BASE=golang:1.18-alpine3.16
FROM ${BASE} AS builder
WORKDIR /app-service-mqtt-go

ARG ALPINE_PKG_BASE="make git gcc libc-dev libsodium-dev zeromq-dev"
ARG ALPINE_PKG_EXTRA=""

RUN apk add --update --no-cache ${ALPINE_PKG_BASE} ${ALPINE_PKG_EXTRA}

COPY go.mod vendor* ./
RUN [ ! -d "vendor" ] && go mod download all || echo "skipping..."

COPY . .
ARG MAKE="make build"
RUN $MAKE

#final stage
FROM alpine:3.16
LABEL license='SPDX-License-Identifier: Apache-2.0' \
  copyright='Copyright (c) 2022: Intel'
LABEL Name=app-service-mqtt-go Version=${VERSION}

# dumb-init is required as security-bootstrapper uses it in the entrypoint script
RUN apk add --update --no-cache ca-certificates zeromq dumb-init
ENV PKG_CONFIG_PATH=/usr/lib/pkgconfig:/usr/local/lib/pkgconfig

COPY --from=builder /app-service-mqtt-go/app-service-mqtt-go /
COPY --from=builder /app-service-mqtt-go/res /res
COPY  LICENSE /

# TODO: set this port appropriatly as it is in the configuation.toml
EXPOSE 59740

ENTRYPOINT ["/app-service-mqtt-go"]
CMD ["-cp=consul.http://edgex-core-consul:8500", "--registry"]
```

**Reply by lenny-goodell** [reply]
> Does the Consul part also work for custom application services or only for the App Service Configurable?

Works will all services that use the `-cp` command line flag

You are using the latest released SDK version v2.3.1 and the Levski release Edgex services, correct?

Please check your service log file to `Info` messages about use of the config provider. You  should see a message similar to these:
```
Using Configuration provider (consul) from: http://edgex-core-consul:8500 with base path of edgex/appservices/2.0/<you-service-key>
Configuration has been pushed to into Configuration Provider (5 envVars overrides applied)
```

**Reply by majidalikhan8baader** [reply]
Thank you for your patience and the replies. As it seems the configuration provider was set wrong in the compose file and was not being set. Your current input related to the logs led me to find out about the issue.
